### ISSUE TITLE
User given ColumnDataSource

### ISSUE THREAD

**Issue by ahartikainen** [arviz-devs/arviz#951]
To enable linked brushing, we need an option for a user to input ColumnDataSource object which contains the needed data. This will also enable to combine different plot types.

We need to be explicit what different kinds of ColumnDataSources we need. At least we have

- mcmc sample
- density (from mcmc sample)
- observed (no chain, draw)

Ideally we would like to have some default fields for good hover tooltip. (Like information for draw, chain, variable name(?); dims +  coords)

**Comment by amukh18** [comment]
I would like to work on this issue

**Comment by amukh18** [comment]
@ahartikainen Could you please give me a brief overview of how the data would be entered?

**Comment by amukh18** [comment]
So the output should be an InferenceData object which we pass to the plotting functions ?

**Comment by ahartikainen** [comment]
I will need to show you one example. I think I have that function somewhere (inferencedata to cds)

**Comment by amukh18** [comment]
This one ?:https://github.com/arviz-devs/arviz/pull/981 I am currently going through this

**Comment by ahartikainen** [comment]
Yes, that one.

I think we could see if that could be used (or is it already used) in the bokeh plotting function.

If it is already used, we could add a possibility include cds in some dictionary/argument (probably in backend_kwargs) and if present, use it instead of CDS created in the function.

Let's think how to handle density-based plots later.

**Comment by amukh18** [comment]
I am a little confused. I was looking at the bokeh backend file for plot_trace and I think there are already some lines of code that convert the InferenceData object to ColumnDataSource in that file itself.

**Comment by ahartikainen** [comment]
Yeah, this is probably much more involved thing that it seems at first, I think I will maybe implement this.

**Comment by amukh18** [comment]
@ahartikainen Okay

**Comment by amukh18** [comment]
@ahartikainen Or Is there any way I could implement at least part of this? I think it would help me learn about Arviz's backend system better

**Comment by ahartikainen** [comment]
Yes, we can implement this together. Let me give you instructions where we start in a few hours

**Comment by amukh18** [comment]
Thank you! I look forward to working on this issue

**Comment by ahartikainen** [comment]
Ok, I did some inspection for our plotting code, and it is certainly not ready for this feature.

I think we really need to do some serious rewriting to enable this.

I think it is better to focus on other issues for now.

cc @OriolAbril

**Comment by amukh18** [comment]
I understand. I will continue my work on the other issues.
